components/google_cloud/google_cloud.app.mjs

```diff
@@ -106,6 +131,65 @@ export default {
     storageClient() {
       return new Storage(this.sdkParams());
     },
+    instancesClient() {
+      return new InstancesClient(this.sdkParams());
+    },
+    zoneOperationsClient() {
+      return new ZoneOperationsClient(this.sdkParams());
+    },
+    zonesClient() {
+      return new ZonesClient(this.sdkParams());
+    },
+    async listVmInstancesByZone(zone) {
+      const instancesClient = this.instancesClient();
+      const sdkParams = this.sdkParams();
+      const [
+        instances,
+      ] = await instancesClient.list({
+        project: sdkParams.projectId,
+        zone,
+      });
+      return instances;
+    },
+    async listZones() {
+      const zonesClient = this.zonesClient();
+      const sdkParams = this.sdkParams();
+      const [
+        zones,
+      ] = await zonesClient.list({
+        project: sdkParams.projectId,
+      });
+      return zones;
+    },
+    async switchInstanceBootStatus(zone, instance, newStatus) {
+      if (newStatus !== "start" && newStatus !== "stop") {
```

Comment:
If you ever have more values to check for, you can do:
```suggestion
      if (!["start", "stop"].includes(newStatus)) {
```

Reply:
good call